### ISSUE TITLE
How to use ncu -ua but lock a few dependency version at the same time?

### ISSUE THREAD

**Issue by kopax** [tjunnone/npm-check-updates#481]
Prior all I would like to thank you for making this OSS.

We have serious issues with all our small projects that need to have a fixed version for a certain dependency.

I am a module writer and we do not want to fix version for our users in our `package.json`, but we want to prevent upgrade voluntary on our side.

Most of the time, when we build something **on top of**, we want to stick to the dependency set by the core one. 

We found `ncu -ua` to be an excellent command when we do not have a fixed version in our package, but we must proceed to a manual edition of `package.json` when we do have fixed version.

Is there a way using a file or whatever, to set a list of ignored dependency when using `ncu -ua` ?

That would save us a lot of time.

**Comment by raineorshine** [comment]
Yes! Add a `.ncurc` file with a `reject` option as described here:

https://github.com/tjunnone/npm-check-updates#configuration-files

**Comment by kopax** [comment]
Great exactly what we were looking for.  Thanks a lot!

**Comment by kopax** [comment]
`ncu -a`

**Comment by raineorshine** [comment]
Let me just get some clarification so I can troubleshoot. When you say "locked dependencies", do you mean the ones that are specified with the `reject` option, or are you referring to something else?

Also you could try `"upgradeAll: true"`, as that is the long option for `-a`.

**Comment by kopax** [comment]
Yes, in my sentence, `locked` is `reject`. 

What is the difference between:

1. `ncu -u`
1. `ncu -ua`
1. `ncu -a`

To me, I have found that `-ua` ignore `reject`, `a` use them, `-u` alone doesn't write the package.json

**Comment by raineorshine** [comment]
Thank you for the clarification. I will investigate to ensure that this feature will work as intended in the upcoming v3 release. I suggest looking at #438 for the new behavior of -a. It has been very confusing that -a and -ua do the same thing, so this is being changed in v3. A patch will not likely be made to v2. I will report here when v3 is published!

**Comment by kopax** [comment]
Great thanks, I agree with #438.

I've made this base on your advice: https://rollup-umd.github.io/ncu/ , It target modules (not apps which need to use a proper lock file)

Basically, it will allow 1 depth of ncu `reject` options incoming from `node_modules` using `keywords` in the dependency `package.json`.

This way, using that module and ncu, we can reuse the reject configuration for new modules projects that use it.

It works but for some weird reason, [this](https://github.com/rollup-umd/ncu/blob/master/src/index.js#L156) is logging twice, while it is run only once by my code base. We use it as it is but In case you would have an explanation for this:

```bash
Auto configuration with @bootstrap-styled/ra-ui
Auto configuration with @yeutech-lab/test-polyfill
Rejected: ra-core, ra-ui-materialui, react-admin, classnames, date-fns, inflection, lodash, node-polyglot, query-string, ra-language-english, react-redux, react-router, react-router-dom, react-router-redux, recompose, redux, redux-form, redux-saga, reselect, @material-ui/core, @material-ui/icons, autosuggest-highlight, react-dropzone, react-headroom, react-autosuggest, material-ui-chip-input, papaparse, whatwg-fetch will be ignored by npm-check-updates.
Auto configuration with @bootstrap-styled/ra-ui
Auto configuration with @yeutech-lab/test-polyfill
Rejected: ra-core, ra-ui-materialui, react-admin, classnames, date-fns, inflection, lodash, node-polyglot, query-string, ra-language-english, react-redux, react-router, react-router-dom, react-router-redux, recompose, redux, redux-form, redux-saga, reselect, @material-ui/core, @material-ui/icons, autosuggest-highlight, react-dropzone, react-headroom, react-autosuggest, material-ui-chip-input, papaparse, whatwg-fetch will be ignored by npm-check-updates.
Using /home/dka/workspace/api/pm-products/package.json
```

**Comment by raineorshine** [comment]
v3 published: https://github.com/tjunnone/npm-check-updates/releases/tag/v3.0.0. Please create a new issue for any bugs you find! Thanks!

**Comment by kopax** [comment]
Thanks I will look at it soon.
